### ISSUE TITLE
PK should not be generated with random_bytes

### ISSUE THREAD

**Issue by kpp** [TokTok/c-toxcore#1169]
https://github.com/TokTok/c-toxcore/blob/2abc00693cc99f277c7c8fe1a201ec33becb01f1/toxcore/DHT.c#L2743..L2751

Since you have a function named `public_key_valid`, PK from `random_bytes` generates an invalid PK each 1/2 attempt.

```c
void random_bytes(uint8_t *data, size_t length)
{
    randombytes(data, length); // from NaCl
}

bool public_key_valid(const uint8_t *public_key)
{
    if (public_key[31] >= 128) { /* Last bit of key is always zero. */
        return 0;
    }

    return 1;
}
```

**Comment by iphydf** [comment]
Thanks. The keys generated there aren't real keys, they are random keys to seed the DHT search list.

This issue still makes sense to fix, but it's not breaking anything. We never actually find or even want to find these keys.

**Comment by kpp** [comment]
> We never actually find or even want to find these keys.

Actually DHT sends `NodesRequest` with PKs of fake friends to fill the Kademlia tree faster. It does not break anything, but it smells a little.

**Comment by iphydf** [comment]
Correct, we send requests for them, but we don't want to find them. I agree it smells and we should fix that. The correct fix involves stronger typing and ensuring that values of type "Public_Key" are never invalid.
